gobblin-temporal/src/main/java/org/apache/gobblin/temporal/joblauncher/GobblinTemporalJobLauncher.java

```diff
@@ -113,7 +124,61 @@ protected void handleLaunchFinalization() {
 
   @Override
   protected void executeCancellation() {
-    log.info("Cancel temporal workflow");
+    if (this.workflowId == null) {
+      log.info("Cancellation of temporal workflow attempted without submitting it");
+      return;
+    }
+
+    log.info("Cancelling temporal workflow {}", this.workflowId);
+    try {
+      WorkflowStub workflowStub = this.client.newUntypedWorkflowStub(this.workflowId);
+
+      // Describe the workflow execution to get its status
+      DescribeWorkflowExecutionRequest request = DescribeWorkflowExecutionRequest.newBuilder()
+          .setNamespace(this.namespace)
+          .setExecution(workflowStub.getExecution())
+          .build();
+      DescribeWorkflowExecutionResponse response = workflowServiceStubs.blockingStub().describeWorkflowExecution(request);
+
+      WorkflowExecutionStatus status;
+      try {
+        status = response.getWorkflowExecutionInfo().getStatus();
+      }
+      catch (Exception e) {
+        log.warn("Exception occurred while getting status of the workflow " + this.workflowId
+            + ". We would still attempt the cancellation", e);
+        workflowStub.cancel();
+        log.info("Temporal workflow {} cancelled successfully", this.workflowId);
```

Comment:
if we intend to cancel anyway, irrespective of whether we get the status, it would be simpler to call cancel directly without describing or checking the workflow status

Reply:
We cancel here only if we catch an exception while getting the status. If we get the status successfully, this piece of code will not be executed.

Reply:
I meant that why do we need to check workflow status, as we are calling cancel if fetching workflow status request fails as well? Can we just call cancel directly(or are there any issues with that) as it would make the code simpler

Reply:
If cancellation is idempotent and doesn't change the status of a job that is already finished, I agree with the point that we should just always cancel and check for the result

Reply:
Canceling a completed job will throw an exception, so it is better if we check the status and then attempt to cancel it. It may happen that job completed between fetching status and attempting to cancel it, but that should be very rare.